### PR TITLE
Fix for network error handler

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

We were using invalid error handler. 
As result error message was missing.

### CODE REVIEW

File: packages/http/httpDataService.js [hunk 1]
```diff
@@ -9,18 +9,19 @@ HttpDataService.prototype.request = function(relativeUrl, httpConfig) {
...
-        console.error('Non-OK response with status ' + response.status);
-        console.error(response);
-        throw new Error(response);
-      }
-      return response.data;
+    return new Promise(function(resolve, reject) {
+      self.$http(httpConfig).then(function successCallback(response) {
+        if (response.status !== 200) {
+          console.error(response);
+          reject(new Error(response.status));
```
Comment (JameelB) [inline packages/http/httpDataService.js]: Is this supposed to be returned?

File: packages/http/httpDataService.js [hunk 2]
```diff
@@ -9,18 +9,19 @@ HttpDataService.prototype.request = function(relativeUrl, httpConfig) {
...
+        }
+        return resolve(response.data);
+      }, function errorCallback(response) {
+        error = new Error(error);
+        console.error(error);
+        reject(error);
```
Comment (JameelB) [inline packages/http/httpDataService.js]: Same as above

File: packages/http/httpDataService.js [hunk 3]
```diff
@@ -9,18 +9,19 @@ HttpDataService.prototype.request = function(relativeUrl, httpConfig) {
...
-        console.error('Non-OK response with status ' + response.status);
-        console.error(response);
-        throw new Error(response);
-      }
-      return response.data;
+    return new Promise(function(resolve, reject) {
+      self.$http(httpConfig).then(function successCallback(response) {
+        if (response.status !== 200) {
+          console.error(response);
+          reject(new Error("Network request failed with invalid status", errror.status));
```
Comment (JameelB) [inline packages/http/httpDataService.js]: does this need to be returned?

Comment (JameelB) [reply packages/http/httpDataService.js]: `error.status` should be `response.status`?

File: packages/http/httpDataService.js [hunk 4]
```diff
@@ -9,18 +9,19 @@ HttpDataService.prototype.request = function(relativeUrl, httpConfig) {
...
+        }
+        return resolve(response.data);
+      }, function errorCallback(error) {
+        console.error(error);
+        error = new Error(error);
+        reject(error);
```
Comment (JameelB) [inline packages/http/httpDataService.js]: should this also be returned?

Comment (wtrocki) [reply packages/http/httpDataService.js]: this is using constructor promise. But no hard to return.

**Comment by wtrocki** [comment]
Steps to reproduce:
1. Launch application
2. Stop server
3. See console. 

More specific error handlers should be added  on top of that (for example dialog if error happened)

**Review by JameelB — APPROVED** [review]
verified locally

**Pull request merged**